pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -574,6 +608,71 @@ func (cm *containerManagerImpl) setFsCapacity() error {
 	return nil
 }
 
+// TODO: move the GetResources logic to PodContainerManager.
+func (cm *containerManagerImpl) GetResources(pod *v1.Pod, container *v1.Container, activePods []*v1.Pod) (*kubecontainer.RunContainerOptions, error) {
+	opts := &kubecontainer.RunContainerOptions{}
+	// Gets devices, mounts, and envs from device plugin handler.
+	glog.V(3).Infof("Calling devicePluginHandler AllocateDevices")
+	// Maps to detect duplicate settings.
+	devsMap := make(map[string]string)
+	mountsMap := make(map[string]string)
+	envsMap := make(map[string]string)
+	allocResps, err := cm.devicePluginHandler.Allocate(pod, container, activePods)
```

Comment:
Ideally, allocation of resources would happen as part of pod creation so as to fail pods during admission whose extended resource request cannot be satisfied by the kubelet due to any reason.
For example, a device may have just turned unhealthy and the scheduler may not have noticed it. Note there can be multiple entities scheduling pods on to nodes.
Node level admission control is the last line of defence against state propagation issues in k8s. 
If not for allocation during pod admission, I'd at-least like to see an admission handler (we have quite a few of them), that validates that devices are available to satisfy a pod's request. Essentially a dry run of the logic here.

Reply:
Agree that sounds a more ideal handling, which probably will be addressed after 1.8 when we try to move the GetResources logic to PodContainerManager.

Reply:
Can you file an issue with the background information just that it doesn't get lost? You can assign the issue to yourself since you plan on fixing it.

Reply:
+1 on what @vishh said.